compiler/infra/SimpleRegex.cpp

```diff
@@ -475,10 +475,8 @@ bool SimpleRegex::match(
       bool isCaseSensitive)
    {
    TR::Compilation *comp = TR::comp();
-   TR::StackMemoryRegion stackMemoryRegion(*comp->trMemory());
```

Comment:
This changes the memory lifetime of the data returned from signature and, possibly regex - why the change?

Reply:
Didn't realize the destructor will unregister the region, I was thinking it's unused instance, will revert the change and update.